### PR TITLE
refactor(connector-fabric-socketio): migrate to fabric-all-in-one from fabric14-fabcar-testnet

### PR DESCRIPTION
**Pull request by outSH**
\- Remove obsolete fabric14\-fabcar\-testnet fabric14\-testnet, use only fabric\-all\-in\-one for all purposes for all test / sample app purposes.
\- Add helper scripts from fabric\-sample that are used during cartrade setup phase \(I've decided it's better to keep them in repo than increasing the size of fabric\-all\-in\-one image\).
\- Upgrade of fabcar network \(from 1.4.1 to 1.4.8\) triggered multiple changes in fabric\-socketio validator, since fabcar nodes \(peers, orderer\) use TLS and require different setup, additionaly done some refactors and fixes in the validator.
\- Update the cartrade and discounted\-cartrade sample apps to work with the new setup, refreshed scripts can be used to ease up the setup process.

Closes: \#1753
Signed\-off\-by: Michal Bajer <michal.bajer@fujitsu.com>

Please note that I'll work on containerizing fabric\-socketio validator and cartrade sample, for this reason I didn't put too much effort in documenting changes from this commit \(since most of it will change very soon\).
Also, docker startup scripts will probably be replaced by sample\-specific docker\-compose.

Depends on #1696

### CODE REVIEW

**Comment by outSH** [comment]
@izuru0 @takeutak @petermetz 
Please review

File: tools/docker/fabric-all-in-one/run-fabric-network.sh [hunk 1]
```diff
@@ -21,9 +21,6 @@ function main()
   tar -cC '/etc/hyperledger/fabric/fabric-ccenv/' . | docker load
   tar -cC '/etc/hyperledger/fabric/fabric-tools/' . | docker load
   tar -cC '/etc/hyperledger/fabric/fabric-ca/' . | docker load
-  tar -cC '/etc/hyperledger/fabric/fabric-couchdb/' . | docker load
```
Comment (petermetz) [inline tools/docker/fabric-all-in-one/run-fabric-network.sh:24]: @outSH Why move the couchdb image loading to the 2.x branch of the if condition? Does 1.4 not support CouchDB? (I don't know, I haven't looked it up)

**Comment by petermetz** [comment]
@outSH Are the files in the `./tools/docker/fabric-all-in-one/fabcar-cli-1.4/` directory a copy paste of the ones in the fabric samples repo?

File: tools/docker/fabric-all-in-one/run-fabric-network.sh (hunk 1 above)
Comment (outSH) [reply tools/docker/fabric-all-in-one/run-fabric-network.sh:24]: @petermetz No, Dockerfile for 1.4 doesn't contain couchdb image and container was failing for this reason :)

**Comment by outSH** [comment]
> Are the files in the ./tools/docker/fabric-all-in-one/fabcar-cli-1.4/ directory a copy paste of the ones in the fabric samples repo?

@petermetz No, I described this in README in this dir, I've done some minor changes to access connection info from local directory, also changed formatting a little

File: tools/docker/fabric-all-in-one/run-fabric-network.sh (hunk 1 above)
Comment (petermetz) [reply tools/docker/fabric-all-in-one/run-fabric-network.sh:24]: @outSH Ouch, got it. Thank you for the fix!

**Review by petermetz — APPROVED** [review]
> > Are the files in the ./tools/docker/fabric-all-in-one/fabcar-cli-1.4/ directory a copy paste of the ones in the fabric samples repo?

@petermetz No, I described this in README in this dir, I've done some minor changes to access connection info from local directory, also changed formatting a little 

@outSH Would it make sense to contribute those minor changes back to the fabric sample repo and then use it from there instead?

**Comment by petermetz** [comment]
> Depends on #1696 

@outSH Unfortunately the robot only sees this syntax when it's placed in the PR description, not in the comments afterwards. (I edited the PR description just now so that the robot sees it)

**Comment by outSH** [comment]
> Would it make sense to contribute those minor changes back to the fabric sample repo and then use it from there instead?

Well, hard-coded path could be replaced with argument (or adjustable by env variable), and although it doesn't seem relevant in any way to fabric-samples itself, I think it could pass, but I'll think about it later - right now I'm working on putting cartrade in a container, and will possibly introduce some changes to these scripts in follow-up commit.

> Unfortunately the robot only sees this syntax when it's placed in the PR description, not in the comments afterwards. (I edited the PR description just now so that the robot sees it)

Didn't know that, thanks! I will put the dependencies in PR description from now on :)

**Review by izuru0 — APPROVED** [review]
LGTM

**Comment by petermetz** [comment]
> > Would it make sense to contribute those minor changes back to the fabric sample repo and then use it from there instead?
> 
> Well, hard-coded path could be replaced with argument (or adjustable by env variable), and although it doesn't seem relevant in any way to fabric-samples itself, I think it could pass, but I'll think about it later - right now I'm working on putting cartrade in a container, and will possibly introduce some changes to these scripts in follow-up commit.
> 

@outSH  Sounds good to me, thanks again!

**Pull request merged**